controllers/nodehealthcheck_controller.go

```diff
@@ -312,7 +321,9 @@ func (r *NodeHealthCheckReconciler) Reconcile(ctx context.Context, req ctrl.Requ
 			return result, err
 		}
 		for _, remediationCR := range remediationCRs {
-			if deleted, err := resourceManager.DeleteRemediationCR(&remediationCR, nhc); err != nil {
+			deleted, leaseRequeueIn, err := resourceManager.DeleteRemediationCR(&remediationCR, nhc, remediationCRs)
```

Comment:
>my initial thought when seeing this is: when I delete a CR, how can this result in a requeue?
Delete is not necessarily successful, IIUC in case it's not, requeue makes sense

Reply:
>also, why does it need the complete CR list? Maybe I miss something, but this looks very unintuitive to me thinking

Initially it was done so  the shortest requeue will be selected among remediations, but I agree you are correct - I'll update the code

Reply:
> Delete is not necessarily successful, IIUC in case it's not, requeue makes sense

in this case an error should be returned